code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ImageOnlyTemplateEventHandler.kt

```diff
@@ -0,0 +1,79 @@
+/*
+  Copyright 2025 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.messaging
+
+import com.adobe.marketing.mobile.MessagingEdgeEventType
+import com.adobe.marketing.mobile.aepcomposeui.ImageOnlyUI
+import com.adobe.marketing.mobile.aepcomposeui.UIAction
+import com.adobe.marketing.mobile.aepcomposeui.UIEvent
+import com.adobe.marketing.mobile.aepcomposeui.state.ImageOnlyCardUIState
+import com.adobe.marketing.mobile.services.Log
+import com.adobe.marketing.mobile.services.ServiceProvider
+
+/**
+ * Image Only Template Event Handler for content card events.
+ *
+ * @property callback An optional callback to invoke when a content card event occurs.
+ */
+internal class ImageOnlyTemplateEventHandler(private val callback: ContentCardUIEventListener?) :
+    MessagingEventHandler(callback) {
```

Comment:
If `MessagingEventHandler` is internal, are we able to parametrize it to accept a state and a template and provide typecasted values ? This will allow you to move the `handleEvent` into the base class and call super.handleEvent() here for display, dismiss and interact.

```
MessagingEventHandler(callback)<Template, State> {

   fun handleEvent(....) {
         val ui = getUI()
        val state = getState()
        when(event) {
            .... 
            ....
          }

   }
   abstract fun getUI() : T
   abstract fun getState() : T
}
```